ompi/mca/osc/pt2pt/osc_pt2pt_data_move.c

```diff
@@ -1660,23 +1687,45 @@ static int ompi_osc_pt2pt_callback (ompi_request_t *request)
 
     osc_pt2pt_gc_clean (module);
 
-    ompi_osc_pt2pt_frag_start_receive (module);
-
-    /* put this request on the garbage colletion list */
-    osc_pt2pt_gc_add_request (module, request);
+    rc = ompi_osc_pt2pt_receive_repost (recv);
 
     OPAL_OUTPUT_VERBOSE((50, ompi_osc_base_framework.framework_output,
-                         "finished posting receive request"));
+                         "finished posting receive request. rc: %d", rc));
 
     return OMPI_SUCCESS;
 }
 
 int ompi_osc_pt2pt_frag_start_receive (ompi_osc_pt2pt_module_t *module)
 {
-    module->frag_request = MPI_REQUEST_NULL;
-    return ompi_osc_pt2pt_irecv_w_cb (module->incoming_buffer, mca_osc_pt2pt_component.buffer_size + sizeof (ompi_osc_pt2pt_frag_header_t),
-                                     MPI_BYTE, OMPI_ANY_SOURCE, OSC_PT2PT_FRAG_TAG, module->comm, &module->frag_request,
-                                     ompi_osc_pt2pt_callback, module);
+    int rc;
+
+    module->recv_frag_count = mca_osc_pt2pt_component.receive_count;
+    if (module->recv_frag_count < 0) {
```

Comment:
@hjelmn I don't see where/how this count can ever be negative. Am I missing a intialization somewhere? I searched and couldn't find it. So this might be dead code.